### PR TITLE
Add collection logo

### LINKED ISSUE #656: Improvement - Ability to add an image as a collection logo
**User Story**

As a user
I want to add a logo when creating a collection
So that my collection can be distinguishable 

**Background**
This aim of this story is to enable users to add a logo/image when creating a collection. The image will display as the the collection cover on the home page, instead of displaying the NFTs within the collection - which is what we currently display. The image will also be displayed within the collection view page. This change should improve performance, as one image is displayed instead of up to four (Alessio/ Onno).
**Maximum file size** = **1mb** Onno/Alessio: Anything over 1mb may lead to issues loading images
**Supported files types** = SVG, PNG, JPEG, GIF, BMP, ICO, WebP, AVIF

**Questions?**
?N/A

**User Journey**

1. User connects wallet
2. User selects 'Create collection'
3. UI displays 'Create new collection' modal
4. User selects and uploads image
5. User enters collection name
6. User enters collection description
7. User clicks ‘Create’ button
8. User signs transaction (JS Wallet pop-up)
9. UI provides transaction pending confirmation (notification toast)
10. Transaction confirmed on-chain - Purchase successful 
11. UI provides action confirmation (notification toast)
12. Collection opens: the collection name, description and logo are displayed

#### Mockups
**To be added**

**Acceptance Criteria**

**Scenario 1 -  User clicks create collection**
Given a user is on the homepage
When they select the ‘create collection’ button
Then a modal should be displayed
And it should allow them to select an image to upload, enter a description and a name

**Scenario 2 -  (Success) User uploads a supported file type under limit**
Given a user is attempting to create a new collection
When the a user selects a  file to import that is either: SVG, PNG, JPEG, GIF, BMP, ICO, WebP, AVIF
And the file size is 1mb or below
Then the file should be uploaded successfully 

**Scenario 3 - User uploads an unsupported file type**
Given a user is attempting to create a new collection
When the a user selects a  file to import that is not either: SVG, PNG, JPEG, GIF, BMP, ICO, WebP, AVIF
Then the file should be not be uploaded
And the user should be presented with the following error 'File format not supported'
And the user should be able to close the error message to return to the upload screen.

**Scenario 3 - User uploads a supported file type that exceeds file size limit**
Given a user is attempting to create a new collection
When the a user selects a  file to import that is either: SVG, PNG, JPEG, GIF, BMP, ICO, WebP, AVIF
And the file size is above 1mb
Then the file should be not be uploaded
And the user should be presented with the following error 'File size exceeded'
And the user should be able to close the error message to return to the upload screen.

**Scenario 4 -  Happy path 1 - User creates a collection with image**
Given a user is attempting to mint an NFT
And they have successfully uploaded an image
And they have entered a collection name
And they have entered a collection title
When they select the 'create' button
And the transaction is confirmed on-chain
Then the logo that was uploaded should be displayed on collection page
And on the homepage as the collections cover art

**Scenario 5 -   Happy path 2 - User creates a collection with no image**
Given a user is attempting to mint an NFT
And they have not uploaded an image
And they have entered a collection name
And they have entered a collection title
When they select the 'create' button
And the transaction is confirmed on-chain
Then a the Altair logo should be displayed on collection page
And on the homepage as the collections cover art

### PR DESCRIPTION
**Pull request by alessioscalici**
### Description

This pull request:
- Allows adding a logo to a collection
- If present, shows the logo in the collection detail page
- If present, shows the logo in the collection list page

Closes #656

### Approvals

- [ ] Dev
- [ ] Dev
- [ ] Designer
- [ ] Product

### Screenshots

Screenshot 2022-03-11 at 14 35 27
Screenshot 2022-03-11 at 14 36 58
Screenshot 2022-03-11 at 14 37 20

### CODE REVIEW

File: nft-studio/src/components/CreateCollectionDialog.tsx [hunk 1]
```diff
@@ -112,6 +117,17 @@ export const CreateCollectionDialog: React.FC<{ open: boolean; onClose: () => vo
             onChange={(e) => setDescription(e.target.value)}
             disabled={fieldDisabled}
           />
+          <Stack gap={1}>
+            <Text variant="label1">Upload collection logo (JPEG, SVG, PNG, or GIF up to 1 MB)</Text>
```
Comment (armandomistral) [inline nft-studio/src/components/CreateCollectionDialog.tsx:121]: Suggesting to not use a verb in a label. Let's change to `Collection logo (JPEG, SVG, PNG, or GIF up to 1 MB)`

File: nft-studio/src/components/CreateCollectionDialog.tsx [hunk 2]
```diff
@@ -112,6 +117,17 @@ export const CreateCollectionDialog: React.FC<{ open: boolean; onClose: () => vo
...
+            <FileUpload
+              onFileUpdate={(file) => setLogoFile(file)}
+              onFileCleared={() => setLogoFile(undefined)}
+              placeholder="Add file"
```
Comment (armandomistral) [inline nft-studio/src/components/CreateCollectionDialog.tsx:125]: Let's change the label of the placeholder to `Upload image`

**Review by armandomistral — CHANGES_REQUESTED** [review]
Looks good to me! Suggested 2 label changes. Thanks for having me review this...

File: nft-studio/src/components/CreateCollectionDialog.tsx (hunk 1 above)
Comment (alessioscalici) [reply nft-studio/src/components/CreateCollectionDialog.tsx:121]: I copied this from Figma. I'll change it, but please next time review the design before putting in the ticket.

**Pull request closed without merging**